### ISSUE TITLE
[SIP-2] Deprecate support for python <3.6

### ISSUE THREAD

**Issue by mistercrunch** [apache/incubator-superset#5565]
# SIP-2 Deprecate support for python <3.6

## Motivation
There's extra cost in maintaining multiple versions of python:
* longer build times (travis build matrix does 2.7 and 3.x now)
* can't take full advantage of nice py3 features like async/await, f-strings and much more
* intricate bugs

## Proposed Change
* remove buid env from tox build matrix
* remove <py3.6 specific code, like `from __future__` imports
* start recommending f-strings instead of `str.format()`
* add notes/instructions in `UPDATE.md`
* update the installation docs

## New or Changed Public Interfaces
People not running >=py3.6 will need to change their installation scripts.

## Migration Plan and Compatibility
N/A

## Rejected Alternatives
Supporting py2.7 forever

**Comment by villebro** [comment]
I would much prefer having only >=py3.6.

**Comment by xrmx** [comment]
python3.5 is still the newest python version for some distros like debian stable that would be maintained for a couple of years more at least. So python3 only is fine but don't use the f-strings yet please.

**Comment by mistercrunch** [comment]
I'd recommend people to not use their distro's stock python to deploy any python service though. Also much more here: https://docs.python.org/3/whatsnew/3.6.html

```
PEP 498, formatted string literals.
PEP 515, underscores in numeric literals.
PEP 526, syntax for variable annotations.
PEP 525, asynchronous generators.
PEP 530: asynchronous comprehensions.
```

**Comment by ottomata** [comment]
> I'd recommend people to not use their distro's stock python to deploy any python service though

Disagree with this :)  The Wikimedia foundation has strict rules about where we get production packages from for security reasons. We've decide to trust Debian and to default to using their packages  when available.

I'd be totally fine with dropping Python 2 support, but please keep this to Python >= 3.5!

**Comment by williaster** [comment]
Closed by #5985
